### PR TITLE
Blog App: validations and model specs

### PR DESCRIPTION
**Pull request by Koshao-sys**
Hi,

**I worked on the following requirements in this PR;**

1. For the User model:
    - Name must not be blank.
    - PostsCounter must be an integer greater than or equal to zero.
  
2. For the Post model: 
    - Title must not be blank.
    - Title must not exceed 250 characters.
    - CommentsCounter must be an integer greater than or equal to zero.
    - LikesCounter must be an integer greater than or equal to zero.

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+                  text: 'Software development is a crucial aspect of the tech industry', comment_counter: 0,
+                  likes_counter: 0)
+      Post.create(author: new_user, title: 'Lugha', text: 'Unasema Lugha?', comment_counter: 0, likes_counter: 0)
+      Post.create(author: new_user, title: 'Peace Maker', text: 'I come in peace?', comment_counter: 0,
+                  likes_counter: 0)
+    end
+
+    it 'returns zero(0) when the user is created' do
+      expect(new_user.recent_posts.count).to eq 0
+    end
```
Comment (elisha2kyakpo1) [inline spec/models/user_spec.rb]: - [ ] Kindly take a look here and fix the failing test. I have attached an image for your reference.

test

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @Koshao-sys, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ The PR title and description are good and professional.
✔️ No linter errors.
✔️ Descriptive readme file.
👏 Model validations are set up nicely.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  end
+end
+
+RSpec.describe User, type: :model do
+  describe '#recent_posts' do
+    let(:new_user) { User.create(name: 'Kunda', photo: 'https://images.unsplash.com/photo-1633332755192-727a05c4013d', bio: 'Tella') }
+
+    it 'returns zero(0) when the user is created' do
+      expect(new_user.recent_posts.count).to eq 0
+    end
```
Comment (AmaduKamara) [inline spec/models/user_spec.rb]: - [ ] Just as the previous reviewer highlighted, the test is still failing on this end here. Kindly work on it and be sure it passes before requesting a review 👍🏽

Screenshot 2023-06-10 at 12 12 06 PM

**Review by AmaduKamara — CHANGES_REQUESTED** [review]
# Status: Required Changes ♻️ ♻️ ♻️

Hi @Koshao-sys, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✅ You used Gitflow correctly 👍🏽
✅ The Linter checks passes 👍🏽
✅ Good descriptive PR 👍🏽

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by Munsa1 — APPROVED** [review]
Hi @Koshao-sys 
congratulations-5

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Made Requested Changes By Previous Reviewers :heavy_check_mark: 
Tested Validations :heavy_check_mark: 
All Passing Test Cases :heavy_check_mark: 
image


## Optional suggestions
### Important :red_circle: 

- [ ] It would be more professional if you added instructions on how to run tests for your project ReadMe under the `Getting Started` section.
image

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**